### PR TITLE
Bail out on overflow

### LINKED ISSUE #178: overflowing-rsh-6.rs fails when run with -O
The following fails to fail:
```
cargo run --bin miri -- -O tests/compile-fail/overflowing-rsh-6.rs
```
Is this deliberate? It seems to me that overflows are still a bug, and there's no deep reason miri shouldn't also detect them in release mode. That would just be a matter of not ignoring the return value of `intrinsic_overflowing`.

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes #178

Also, remove a FIXME that was already fixed and extend a clarifying comment in memory.rs.

### CODE REVIEW

File: src/eval_context.rs [hunk 1]
```diff
@@ -452,8 +452,14 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             }
 
             BinaryOp(bin_op, ref left, ref right) => {
-                // ignore overflow bit, rustc inserts check branches for us
-                self.intrinsic_overflowing(bin_op, left, right, dest, dest_ty)?;
+                if self.intrinsic_overflowing(bin_op, left, right, dest, dest_ty)? {
+                    // There was an overflow in an unchecked binop.  Right now, we consider this an error and bail out.
+                    // The rationale is that the reason rustc emits unchecked binops in release mode (vs. the checked binops
+                    // it emits in debug mode) is performance, but it doesn't cust us any performance in miri.
```
Comment (bjorn3) [inline src/eval_context.rs]: *cost

File: src/memory.rs [hunk 2]
```diff
@@ -166,7 +166,8 @@ pub struct Memory<'a, 'tcx> {
     /// We mark memory as "packed" or "unaligned" for a single statement, and clear the marking
     /// afterwards. In the case where no packed structs are present, it's just a single emptyness
     /// check of a set instead of heavily influencing all memory access code as other solutions
-    /// would.
+    /// would.  This is simpler than the alternative of passing a "packed" parameter to every
+    /// load/store method.
```
Comment (bjorn3) [inline src/memory.rs]: Double space

**Comment by RalfJung** [comment]
Fixed.

**Comment by RalfJung** [comment]
I fixed another comment.

**Pull request merged**